lib/catalog/survey_compare.rb

```diff
@@ -0,0 +1,25 @@
+module Catalog
+  class SurveyCompare
+    attr_reader :base
+    class << self
+      def changed?(plan)
```

Comment:
@syncrou Do we need to add an issue or a JIRA story that has a method on the service plan that does the comparison during order time to ensure that the base has not changed.

Reply:
Here is one for checking on edit: https://projects.engineering.redhat.com/browse/SSP-904

Here is the story for checking before submitting an order: https://projects.engineering.redhat.com/browse/SSP-905

Reply:
@mkanoor I added the requested stories above to Jira. I was planning on implementing them separate to this PR. I wanted this PR to implement only the backend requirements for the Base64 checking without applying it to the endpoints.